tools/eslint-rules/rules/ngrx-fail-action-must-initialize-error.spec.ts

```diff
@@ -0,0 +1,239 @@
+            error = 'some error'; // Property initializer
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Constructor parameter',
+      code: `
+        export interface ErrorAction {
+          error: Object;
+        }
+
+        export class SearchProductsFail implements ErrorAction {
+            readonly type = SEARCH_PRODUCTS_FAIL;
+            constructor(public error: any) {} // Constructor parameter
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Constructor assignment',
+      code: `
+        export interface ErrorAction {
+          error: Object;
+        }
+
+        export class SearchProductsFail implements ErrorAction {
+            readonly type = SEARCH_PRODUCTS_FAIL;
+            error: Object;
+            constructor(public payload: any) {
+                this.error = payload; // Constructor assignment
+            }
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Constructor body assignment',
+      code: `
+        export interface ErrorAction {
+          error: Object;
+        }
+
+        export class SearchProductsFail implements ErrorAction {
+            readonly type = SEARCH_PRODUCTS_FAIL;
+            error: any;
+            constructor() {
+                this.error = 'error'; // Constructor body assignment
+            }
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Class not implementing ErrorAction',
+      code: `
+        export interface ErrorAction {
+          error: Object;
+        }
+
+        export class SearchProductsFail {
+            readonly type = SEARCH_PRODUCTS_FAIL;
+          constructor(public payload: any) {}
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Class with Fail but not implementing ErrorAction',
+      code: `
+        export class LoadProductsFail {
+            readonly type = LOAD_PRODUCTS_FAIL;
+            constructor(public payload: Error) {}
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Multiple interfaces implementation',
+      code: `
+        export interface ErrorAction { 
+          error: Object; 
+        }
+        
+        export interface PayloadAction { 
+          payload: any; 
+        }
+        
+        export class ComplexFail implements ErrorAction, PayloadAction {
+          readonly type = COMPLEX_FAIL;
+          constructor(public payload: any) {
+            this.error = payload;
+          }
+        }
+      `,
+      filename: 'file.ts',
+    },
+    {
+      name: 'Inheritance case',
+      code: `
+        export interface ErrorAction { error: Object; }
+        export class BaseFailAction implements ErrorAction {
+          error = new Error();
+        }
+        export class DerivedFail extends BaseFailAction {
+          readonly type = DERIVED_FAIL;
+        }
+      `,
+      filename: 'file.ts',
+    },
```

Comment:
For being more bullet-proof, let's add also a case with a grandparent (to assert it's not only a shallow parent-check but likely a recursive check)
```suggestion
    {
      name: 'Inheritance case',
      code: `
        export interface ErrorAction { error: Object; }
        export class BaseFailAction implements ErrorAction {
          error = new Error();
        }
        export class DerivedFail extends BaseFailAction {
          readonly type = DERIVED_FAIL;
        }
      `,
      filename: 'file.ts',
    },
    {
      name: 'Grandparent inheritance case',
      code: `
        export interface ErrorAction { error: Object; }
        export class GrandparentFailAction implements ErrorAction {
          error = new Error();
        }
        export class ParentFailAction extends GrandparentFailAction {
          readonly type = PARENT_FAIL;
        }
        export class ChildFailAction extends ParentFailAction {
          readonly type = CHILD_FAIL;
        }
      `,
      filename: 'file.ts',
    },
```